src/stores/QueryStore.ts

```diff
@@ -26,6 +29,9 @@ export interface QueryStoreState {
     readonly currentRequest: CurrentRequest
     readonly maxAlternativeRoutes: number
     readonly routingProfile: RoutingProfile
+    readonly customModelEnabled: boolean
+    readonly customModelValid: boolean
+    readonly customModel: CustomModel | null
```

Comment:
I added three new fields to the app's state: `customModelEnabled` simply tells us whether we are in 'custom model mode' or not (and right now it also determines whether the custom model editor is shown or not). `customModelValid` tells us whether or not the model is valid as it is determined by the editor component. `customModel` is the current custom model. We cannot just set the `customModel` field to null in case it is invalid, because we like to be able to send requests also for invalid models. Note that `customModel` is **not** used as backend for the editor's actual content. This is because if we managed the editor's content using react we would need to manage all of the editor's state (like cursor position, annotations etc.) with react (or at least keep it up to date somehow).  Right now I don't see a real advantage of this.